python/gilded_rose.py

```diff
@@ -7,33 +7,38 @@ def __init__(self, items):
 
     def update_quality(self):
         for item in self.items:
-            if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert":
-                if item.quality > 0:
-                    if item.name != "Sulfuras, Hand of Ragnaros":
-                        item.quality = item.quality - 1
-            else:
-                if item.quality < 50:
-                    item.quality = item.quality + 1
-                    if item.name == "Backstage passes to a TAFKAL80ETC concert":
-                        if item.sell_in < 11:
-                            if item.quality < 50:
-                                item.quality = item.quality + 1
-                        if item.sell_in < 6:
-                            if item.quality < 50:
-                                item.quality = item.quality + 1
-            if item.name != "Sulfuras, Hand of Ragnaros":
-                item.sell_in = item.sell_in - 1
-            if item.sell_in < 0:
-                if item.name != "Aged Brie":
-                    if item.name != "Backstage passes to a TAFKAL80ETC concert":
-                        if item.quality > 0:
-                            if item.name != "Sulfuras, Hand of Ragnaros":
-                                item.quality = item.quality - 1
-                    else:
-                        item.quality = item.quality - item.quality
+            decrease_qual = 1
+
+            if item.name == 'Sulfuras, Hand of Ragnaros':
+                if item.sell_in > 0:
+                    item.sell_in -= 1
+                continue
+
+            if item.name == 'Aged Brie':
+                decrease_qual = -1
+            elif item.name.startswith("Backstage passes"):
+                if item.sell_in == 0:
+                    decrease_qual = item.quality
+                elif item.sell_in <= 5:
+                    decrease_qual = -3
+                elif item.sell_in <= 10:
+                    decrease_qual = -2
                 else:
-                    if item.quality < 50:
-                        item.quality = item.quality + 1
+                    decrease_qual = -1
+            elif item.name.startswith("Conjured"):
+                decrease_qual = 2
+
+            if decrease_qual > 0 and item.sell_in == 0:
+                    decrease_qual *= 2
+
+            if item.quality > 0:
+                item.quality = max(0, item.quality - decrease_qual)
+
+            if item.quality > 50:
+                item.quality = 50
```

Comment:
I think we can make the `item.quality` a one-liner here by using `min(50, max(...))`